Sources/HTTPClient/HTTPClient.swift

```diff
@@ -0,0 +1,8 @@
+import Foundation
+import HTTPTypes
+
+public protocol HTTPClientProtocol {
+  associatedtype Body
```

Comment:
add associatedType Body, Data for `Data`, `ByteBuffer`, `[Int8]`.
https://github.com/apple/swift-http-types/issues/15#issuecomment-1646736121